packages/contracts-bedrock/scripts/upgrades/holocene/scripts/sys-cfg-bundle.sh

```diff
@@ -30,28 +30,54 @@ L1_CHAIN_ID=$(cast chain-id)
 # Copy the bundle template
 cp ./templates/sys_cfg_upgrade_bundle_template.json "$BUNDLE_PATH"
 
+
 # We need to re-generate the SystemConfig initialization call
-# We want to use the exact same values that the SystemConfig is already using, apart from baseFeeScalar and blobBaseFeeScalar.
-# Start with values we can just read off:
+# We want to use the exact same values that the SystemConfig is already using, apart from baseFeeScalar, blobBaseFeeScalar, gasPayingToken, disputeGameFactory.
+# Start with values that are the same regardless of existing SystemConfig version:
 OWNER=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "owner()")
 SCALAR=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "scalar()")
 BATCHER_HASH=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "batcherHash()")
 GAS_LIMIT=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "gasLimit()")
 UNSAFE_BLOCK_SIGNER=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "unsafeBlockSigner()")
 RESOURCE_CONFIG=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "resourceConfig()")
 BATCH_INBOX=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "batchInbox()")
-GAS_PAYING_TOKEN=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "gasPayingToken()(address)")
 L1_CROSS_DOMAIN_MESSENGER_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "l1CrossDomainMessenger()(address)")
 L1_STANDARD_BRIDGE_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "l1StandardBridge()(address)")
 L1_ERC721_BRIDGE_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "l1ERC721Bridge()(address)")
-DISPUTE_GAME_FACTORY_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "disputeGameFactory()(address)")
 OPTIMISM_PORTAL_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "optimismPortal()(address)")
 OPTIMISM_MINTABLE_ERC20_FACTORY_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "optimismMintableERC20Factory()(address)")
 
+# 2.3.0 is the first release to have a gasPayingToken. Therefore we set it to
+# the magic address denoting Ether, since we are upgrading chains which are
+# by definition not custom gas token chains
+GAS_PAYING_TOKEN=0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE
+
+# Handle DisputeGameFactory in a version-specific way:
+SYSTEM_CONFIG_VERSION=$(cast abi-decode "version()(string)" "$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "version()")" | tr -d '"')
```

Comment:
```suggestion
SYSTEM_CONFIG_VERSION=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" 'version()(string)' | tr -d '"')
```
if you specify the return value for `cast call`, it will directly decode it correctly